Week2/homework/exercise4.js

```diff
@@ -0,0 +1,40 @@
+"use strict"
+var books = [
+    {
+        title: "We Should Be All Feminists",
+        author: "Chimamanda Ngozi Adichie",
+        alreadyread: true
+    },
+
+    {
+        title: "My Husband's Secret",
+        author: "Liane Moriarty",
+        alreadyread: true
+    },
+
+    {
+        title: "The Second Sex",
+        author: "Simone de Beauvoir",
+        alreadyread: false
+    }
+]
+
+var i = 0;
+var bookslen = books.length;
+var bookskey = Object.keys(books[i]);
+console.log(bookslen)
+/*this is to run through the books list*/
+for (i = 0; i < bookslen; i++){
+    console.log(books[i])
+}
+/*this is to show book by book and the status if I already read it or not*/
+for (i = 0; i < bookslen; i++){
+    console.log(books[i][bookskey[0]] + " by " + books[i][bookskey[1]])
+    if (books[i][bookskey[2]] === true)
+    console.log("You already read " + books[i][bookskey[0]])
+         else 
+    console.log("You still need to read " + books[i][bookskey[0]])
+    console.log("-----")
+}
```

Comment:
Ah. This is exactly why I mentioned earlier we should use `let` instead of `var`. Ideally, the index(`i`) inside `for` loop shouldn't be reused inside a different for loop as both of them should have their own instances of the index. This would have been achieved if you used `let` instead of `var`.

Also, it's better to declare `i` inside the for loop instead of doing it separately as you do in line 22. This kind of coding again makes me ask the question if you are referring any reading material for JavaScript outside HYF's provided material. Would definitely recommend checking the timestamp of it to make sure it is recent and hence upto date with recent developments in JavaScript.

Reply:
Oh, noooo, I am really suffering from the var disease hahaha. I'll let them away, promise.
About declaring i = 0 inside the loop, I remember I start to do that after week3 but specially in this case, because of the books.length, if I change the code stop working. In that case, can I leave like that or you have a better suggestion?